lib/features/support/server.rb

```diff
@@ -1,86 +1,73 @@
 # frozen_string_literal: true
 
-require 'webrick'
 require 'json'
+require 'singleton'
+require 'webrick'
 require_relative './servlet'
 require_relative './logger'
-
-# This port number is semi-arbitrary. It doesn't matter for the sake of
-# the application what it is, but there are some constraints due to some
-# of the environments that we know this will be used in - namely, driving
-# remote browsers on BrowserStack. The ports/ranges that Safari will access
-# on "localhost" urls are restricted to the following:
-#
-#   80, 3000, 4000, 5000, 8000, 8080 or 9000-9999
-#   [ from https://stackoverflow.com/a/28678652 ]
-#
-MOCK_API_PORT = 9339
+require_relative './request_list'
 
 # Receives and stores requests through a WEBrick HTTPServer
 class Server
-  class << self
-    # Whether the server thread is running
-    #
-    # @return [Boolean] If the server is running
-    def running?
-      @thread&.alive?
-    end
+  include Singleton
```

Comment:
I'm not sure if we need to move to a singleton model.  It doesn't seem to really provide us with anything, but makes accessing the server methods more complex.

Reply:
Yes I think you're right now we've talk about it IRL.  I'll take another look.

Reply:
Now reverted to use Eignenclass approach.  Much better.